internal/transactions/sign.go

```diff
@@ -82,14 +83,33 @@ func sign(
 		return nil, fmt.Errorf("failed to read partial transaction from %s: %v", filenameOrUrl, err)
 	}
 
-	signer, err := state.Accounts().ByName(signFlags.Signer)
+	var signed *flowkit.Transaction
+	var signers []*flowkit.Account
+	tx, err := flowkit.NewTransactionFromPayload(payload)
 	if err != nil {
-		return nil, fmt.Errorf("signer account: [%s] doesn't exists in configuration", signFlags.Signer)
+		return nil, err
 	}
 
-	signed, err := services.Transactions.Sign(signer, payload, globalFlags.Yes)
-	if err != nil {
-		return nil, err
+	//validate all signers
+	for _, signerName := range signFlags.Signer {
+		signer, err := state.Accounts().ByName(signerName)
+		if err != nil {
+			return nil, fmt.Errorf("signer account: [%s] doesn't exists in configuration", signerName)
+		}
+		signers = append(signers, signer)
+	}
+
+	//payer signs last
+	sort.SliceStable(signers, func(i, j int) bool {
+		return signers[i].Address().String() != tx.FlowTransaction().Payer.Hex()
+	})
+
+	for _, signer := range signers {
+		signed, err = services.Transactions.Sign(signer, payload, globalFlags.Yes)
+		if err != nil {
+			return nil, err
+		}
+		payload = []byte(hex.EncodeToString(signed.FlowTransaction().Encode()))
```

Comment:
not sure if we need this

Reply:
`payload` part ? Last signer has to sign the envelope ( which has to include previous signatures )

Reply:
nevermind lol